src/malli/core.cljc

```diff
@@ -2718,6 +2718,66 @@
                                   :re-transformer (fn [_ children] (apply re/alt-transformer children))
                                   :re-min-max (fn [_ children] (reduce -re-alt-min-max {:max 0} (-vmap last children)))})})
 
+(defn -deref-schema [{:keys [type]}]
```

Comment:
I understand the naming, but I fear confusion with `m/deref`. Not sure what to do about that. Docstring?

Reply:
My original intention for this name was to have an overarching `[:deref S]` schema. I couldn't decide whether it should be a simple `reify` or a disjunction of `:delay`/`:promise`/`:future`.

I'm not sure if providing a schema called `:deref` increases or reduces clarity here.